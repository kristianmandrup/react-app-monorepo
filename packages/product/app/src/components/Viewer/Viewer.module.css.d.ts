export const container: string;
export const title: string;
export const commom: string;
export const big: string;
export const medium: string;
export const small: string;
