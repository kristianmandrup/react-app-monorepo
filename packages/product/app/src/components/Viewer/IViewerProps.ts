export interface IViewerProps {
  id: string
  article: any
}
