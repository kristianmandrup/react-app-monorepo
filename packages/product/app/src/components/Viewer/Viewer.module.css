.container {
  max-width: 1280px;
  text-align: center;
  margin: 0 auto;
}

.title {
  text-decoration: underline;
  font-size: 30px;
  margin-top: 10px;
}

.commom {
  text-align: left;
  float: left;
  margin: 15px 15px;
}

.big {
  composes: commom;
}

.medium {
  composes: commom;
}

.small {
  composes: commom;
}

