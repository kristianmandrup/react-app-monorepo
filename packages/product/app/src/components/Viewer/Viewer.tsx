import * as React from 'react'
import ViewerItem from '../ViewerItem/ViewerItem'
import * as styles from './Viewer.module.css'
import { ViewerItemCardType } from '../ViewerItem/ViewerItemCardType'
import { IViewerProps } from './IViewerProps'

class Viewer extends React.Component<IViewerProps, {}> {
  public render() {
    const { id, article } = this.props
    return (
      <section id={id} className={styles.container} >
        <div className={styles.title} >Card Types sample</div>
        <div className={styles.big}>
          <ViewerItem {...article} typeSingleton={ViewerItemCardType.Big} />
        </div>
        <div className={styles.medium}>
          <ViewerItem {...article} typeSingleton={ViewerItemCardType.Medium} />
        </div>
        <div className={styles.small}>
          <ViewerItem {...article} typeSingleton={ViewerItemCardType.Small} />
        </div>
      </section>
    )
  }
}

export default Viewer
