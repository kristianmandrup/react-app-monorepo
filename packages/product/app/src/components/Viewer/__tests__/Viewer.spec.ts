import testComponentHelper from '../../../../../test/unit/helpers/ComponentHelper'
import Viewer from '../Viewer'

describe('<Viewer />', () => {
  const initialProps = {
    className: ''
  }
  const renderComponent = testComponentHelper(Viewer)
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent().getHtml())
        .toMatchSnapshot()
    })
  })
})
