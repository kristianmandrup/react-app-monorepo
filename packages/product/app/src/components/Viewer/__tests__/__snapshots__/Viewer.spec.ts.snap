// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Viewer /> @renders in default state 1`] = `
<section
  className="container"
>
  <div
    className="title"
  >
    Card Types sample
  </div>
  <div
    className="big"
  >
    <ViewerItem
      typeSingleton={
        ViewerItemCardType {
          "dateMinHeight": 18,
          "id": 1,
          "imageHeight": 330,
          "imageWidth": 660,
          "name": "FeaturedBig",
          "summaryFontSize": 16,
          "summaryMaxWords": 160,
          "summaryMinHeight": 58,
          "titleFontSize": 21,
          "titleMaxWords": 110,
          "titleMinHeight": 68,
        }
      }
    />
  </div>
  <div
    className="medium"
  >
    <ViewerItem
      typeSingleton={
        ViewerItemCardType {
          "dateMinHeight": 16,
          "id": 2,
          "imageHeight": 155,
          "imageWidth": 310,
          "name": "FeaturedSmall",
          "summaryFontSize": 14,
          "summaryMaxWords": 125,
          "summaryMinHeight": 68,
          "titleFontSize": 20,
          "titleMaxWords": 50,
          "titleMinHeight": 66,
        }
      }
    />
  </div>
  <div
    className="small"
  >
    <ViewerItem
      typeSingleton={
        ViewerItemCardType {
          "dateMinHeight": 14,
          "id": 3,
          "imageHeight": 100,
          "imageWidth": 200,
          "name": "NormalArticle",
          "summaryFontSize": 12,
          "summaryMaxWords": 115,
          "summaryMinHeight": 75,
          "titleFontSize": 18,
          "titleMaxWords": 35,
          "titleMinHeight": 62,
        }
      }
    />
  </div>
</section>
`;
